sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/models/AmqpMessageBody.java

```diff
@@ -3,14 +3,113 @@
 
 package com.azure.core.amqp.models;
 
+import com.azure.core.util.IterableStream;
+import com.azure.core.util.logging.ClientLogger;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+
 /**
- * Interface representing Amqp Message Body.
+ * This class encapsulates the body of a message. The {@link AmqpMessageBodyType} map to AMQP specification message body
+ * types. Current implementation only support {@link AmqpMessageBodyType#DATA DATA}. Other types will be supported in
+ * future releases.
+ * <b>Amqp message body types:</b>
+ * <ul>
+ * <li><a href="http://docs.oasis-open.org/amqp/core/v1.0/os/amqp-core-messaging-v1.0-os.html#type-data">DATA</a></li>
+ * <li><a href="http://docs.oasis-open.org/amqp/core/v1.0/amqp-core-messaging-v1.0.html#type-amqp-sequence">SEQUENCE</a></li>
+ * <li><a href="http://docs.oasis-open.org/amqp/core/v1.0/os/amqp-core-messaging-v1.0-os.html#type-amqp-value">VALUE</a></li>
+ * </ul>
+ *
+ * <b>Client should test for {@link AmqpMessageBodyType} before calling corresponding get method.Get methods not
+ * corresponding to the type of the body throws exception.</b>
+ *
+ * <p><strong>How to check for {@link AmqpMessageBodyType}</strong></p>
+ * {@codesnippet com.azure.core.amqp.models.AmqpBodyType.checkBodyType}
+ *
+ * @see AmqpMessageBodyType
  */
-public interface AmqpMessageBody {
+public final class AmqpMessageBody {
+    private final ClientLogger logger = new ClientLogger(AmqpMessageBody.class);
+    private AmqpMessageBodyType bodyType;
+
+    private List<byte[]> data;
+
+    AmqpMessageBody() {
```

Comment:
This can be made private as it's not required by any other class within the package.